### PR TITLE
Make `Filters Products by Price` work with `Active Filters` block

### LINKED ISSUE #6243: Make Filters Products by Price work with Active Filters block
## Describe the bug

When the PR https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6146 was worked on and merged, we failed to consider the Active Filters block which shows currently active filters.

## To reproduce

1. Go to Shop page with the Filter Products by Price block & Active Filters block
2. Use the Filter Products by Price block, the page should reload with the filter params now being in the URL too.
4. Notice the Active Filters block does not show the active price filter.

## Expected behavior

I expect the Active Filters block to show the min & max prices as active when they are used.

### PR DESCRIPTION
**Pull request by albarin**
When we render the PHP Classic template with the Filter Products by Price block we need to make sure the Active Filters block is also updated with the `min_price` and `max_price` values from the URL.

Note: clicking on **Clear All** on the `Active Filters` block won't actually clear the price on the slider. That's a known limitation of the `Active Filters` block and will be addressed on here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6138

Fixes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/6243

### Testing

- Add the `Filter Products by Price` block & `Active Filters` block to the **Shop** Page
- Open the **Shop** page
- Use the `Filter Products by Price` block, the page should reload with the filter params now being in the URL too.
- Check the `Active Filters` block shows the active price filter.

### Changelog

> Fix: Make Filters Products by Price work with Active Filters block for the PHP rendered Classic Template.

### CODE REVIEW

File: assets/js/blocks/price-filter/block.js [hunk 1]
```diff
@@ -71,6 +84,8 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 		''
 	);
 
+	const [ hasSetPriceDefault, setHasSetPriceDefault ] = useState( false );
```
Comment (tjcafferkey) [inline assets/js/blocks/price-filter/block.js]: Do you think we should add a comment above this piece of state mentioning that it is to track the defaults we set only on the PHP-rendered Classic Block pages? As its not used for the All Products block.

File: assets/js/blocks/price-filter/block.js [hunk 2]
```diff
@@ -108,6 +123,33 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 		minorUnit: currency.minorUnit,
 	} );
 
+	/**
+	 * Important: For PHP rendered block templates only.
+	 *
+	 * When we render the PHP block template (e.g. Classic Block) we need
+	 * to set the default min_price and max_price values from the URL
+	 */
```
Comment (tjcafferkey) [inline assets/js/blocks/price-filter/block.js:126]: Let's also mention in this comment that we need to set the default state so this filter works alongside the Active Filters block.

**Comment by tjcafferkey** [comment]
This tests well for me but will leave the final review for @tomasztunik seeing as I also worked on this PR 😄

**Review by tomasztunik — APPROVED** [review]
Looks/works good 👌 

Noticed differences as to how filtering is handled on /shop page vs custom built shop page with All Products block and filters added on the side. On /shop page there is full page reload and if user has scrolled view it would jump back to top and sometimes it would have the blocks reload asynchronously (or data fetch would cause placeholder to show up) while on the custom built page it works nice without any reloads or scroll jumps. 

Active filters  also works on All Products approach. It's not perfect there though as the URL params are not updated for the All Products variant.

Just sharing observations, not part of the issue!

(ps fixed conflicting file)

**Pull request merged**